apps/angular/src/app/anime-dashboard/anime-dashboard.component.ts

```diff
@@ -1,28 +1,218 @@
+
+	private readonly subscriptions = new Subscription();
+
+	private animeData$ = new Observable<AnimeData>();
+
+	/** Filtering types Control. */
+	protected readonly filteringTypes = new FormControl<DisplayedAnimeType[] | undefined>(undefined);
+
+	/** Search string Control. */
+	protected readonly searchString = new FormControl<string>('');
+
+	/** Pagination subject. */
+	protected readonly paginationSubject$ = new BehaviorSubject<PaginationConfig>(this.paginationConfig);
+
+	/** Search subject. */
+	protected readonly searchSubject$ = new BehaviorSubject<string>('');
+
+	/** Anime count. */
+	protected readonly animeCount$ = new Observable<number>();
+
+	/** Anime list on page. */
+	protected animeOnPage$ = new Observable<Anime[]>();
 
-	/** All anime. */
-	protected readonly allAnime$ = this.animeService.getAll();
+	/** Total count of Anime. */
+	protected totalAnimeCount$ = new Observable<number>();
+
+	/** @inheritdoc */
+	public ngOnInit(): void {
+
+		this.subscriptions.add(this.filteringTypes.valueChanges.subscribe(
+			() => {
+				if (this.filteringTypes.value == null) {
+					this.typeFilterSubject$.next(undefined);
+					return;
+				}
+
+				const apiKeys = this.filteringTypes.value.map<ApiSideKeyAnimeType>(item => this.animeTypeMapper.map(item));
+				this.typeFilterSubject$.next(apiKeys);
+			},
+		));
+
+		this.subscriptions.add(this.searchString.valueChanges.subscribe(
+			() => {
+				if (this.searchString.value != null) {
+					this.searchSubject$.next(this.searchString.value);
+				}
+			},
+		));
+
+		// Subjects.
+		this.animeData$ = combineLatest([
+			this.sortSubject$,
+			this.paginationSubject$,
+			this.typeFilterSubject$,
+			this.searchSubject$,
+		]).pipe(
+			switchMap(
+				([
+					sortConfig,
+					paginationConfig,
+					typeFilterConfig,
+					searchString,
+				]) => this.animeService.getAll(sortConfig, paginationConfig, typeFilterConfig, searchString),
+			),
+			shareReplay({ bufferSize: 1, refCount: true }),
+		);
+
+		this.animeOnPage$ = this.animeData$.pipe(
+			map(animeList => animeList.pageData),
+		);
+
+		this.totalAnimeCount$ = this.animeData$.pipe(
+			map(animeList => animeList.totalCount),
+		);
+
+		this.subscriptions.add(this.searchSubject$.subscribe(() => {
+			this.resetPageIndex();
+		}));
+
+		this.subscriptions.add(this.typeFilterSubject$.subscribe(() => {
+			this.resetPageIndex();
+		}));
+	}
+
+	private resetPageIndex(): void {
+		this.paginationConfig.pageIndex = 0;
+		this.paginationSubject$.next(this.paginationConfig);
+	}
+
+	/** @inheritdoc */
+	public ngOnDestroy(): void {
+		this.subscriptions.unsubscribe();
+	}
+
+	/**
+	 * Handle page event.
+	 * @param e PageEvent.
+	 */
+	protected handlePageEvent(e: PageEvent): void {
+		// Mapping PageEvent to PaginationConfig.
+		this.paginationConfig.pageIndex = e.pageIndex;
+		this.paginationConfig.pageSize = e.pageSize;
+		this.paginationSubject$.next(this.paginationConfig);
```

Comment:
Looks confusing, why not just do this?
```suggestion
		this.paginationSubject$.next({
			pageIndex: e.pageIndex,
			pageSize: e.pageSize,
		});
```

If you want to access pagination without subscribing, it's a bad practice, because now we have 2 sources of truth. In angular we use rx.